lib/modules/hosts/giphy.js

```diff
@@ -1,27 +1,35 @@
+import { ajax } from '../../environment';
+import logo from '../../images/hosts/giphy-logo.png';
+
 export default {
 	moduleID: 'giphy',
 	name: 'giphy',
 	domains: ['giphy.com'],
-	logo: '//giphy.com/static/img/favicon.png',
-	detect: ({ pathname }) => (/^\/gifs\/(?:\w+-)*(.*?)(\/html5)?$/i).exec(pathname),
-	handleLink(href, [, id, isHtml5]) {
+	logo,
+	detect: ({ pathname }) => (/^(?:\/gifs|\/media|)\/(?:\w+-)*([^/.]*)(\/html5)?.*$/i).exec(pathname),
```

Comment:
For the second point, it was to deal with links to i.giphy.com which have links with the following format: http://i.giphy.com/MhR4xjfLE2eNq.gif